File: python-project-euler-100/p013.py
from Solution import Solution

class P013(Solution):
    def read_numbers(self, file_name):
        numbers = []
        with open(file_name) as file_in:
            for line in file_in:
                numbers.append(line)
        return numbers
    
    def solve(self):
        self.problem_number = 13
        sum = 0
        numbers = self.read_numbers("p013.txt")
    
        for i in numbers:
            sum += int(i);
    
        result = int(str(sum)[0:10])
        return result

def main():
	P013().run()

if __name__ == "__main__":
	main()

File: java-project-euler-100/src/main/java/net/talaatharb/projecteuler100/P013.java
package net.talaatharb.projecteuler100;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.math.BigInteger;
import java.util.ArrayList;
import java.util.List;

/**
 * P013
 * 
 * @author mharb
 *
 */
public class P013 extends Solution {

	/**
	 * Problem solution entry point
	 * 
	 * @param args
	 */
	public static void main(String[] args) {
		new P013().run();
	}

	private List<String> readNumbers(final String fileName) {
		List<String> numbers = new ArrayList<>();

		ClassLoader classLoader = Thread.currentThread().getContextClassLoader();

		InputStreamReader streamReader = new InputStreamReader(classLoader.getResourceAsStream(fileName));
		BufferedReader reader = new BufferedReader(streamReader);
		try {
			for (String line; (line = reader.readLine()) != null;) {
				numbers.add(line);
			}
			reader.close();
			streamReader.close();
		} catch (IOException e) {
			System.err.println(e.getMessage());
		}

		return numbers;
	}

	@Override
	public Number solve() {
		problemNumber = 13;
		BigInteger sum = BigInteger.valueOf(0);
		final List<String> numbers = readNumbers("p013.txt");

		for (String line : numbers) {
			sum = sum.add(new BigInteger(line));
		}
		final long result = Long.parseLong(sum.toString().substring(0, 10));

		return result;
	}

}
